lightning/src/routing/router.rs

```diff
@@ -1267,8 +1307,26 @@ where L::Target: Logger {
 				}
 			}
 		},
-		_ => return Err(LightningError{err: "Routing to blinded paths isn't supported yet".to_owned(), action: ErrorAction::IgnoreError}),
-
+		Payee::Blinded { route_hints, .. } => {
+			if route_hints.iter().all(|(_, path)| &path.introduction_node_id == our_node_pubkey) {
+				return Err(LightningError{err: "Cannot generate a route to blinded paths if we are the introduction node to all of them".to_owned(), action: ErrorAction::IgnoreError});
+			}
+			for (_, blinded_path) in route_hints.iter() {
+				if blinded_path.blinded_hops.len() == 0 {
+					return Err(LightningError{err: "0-hop blinded path provided".to_owned(), action: ErrorAction::IgnoreError});
+				} else if &blinded_path.introduction_node_id == our_node_pubkey {
+					log_info!(logger, "Got blinded path with ourselves as the introduction node, ignoring");
```

Comment:
Hmm, I don't see how that would work if the `max_htlc` of the 1 blinded hint isn't sufficient for the entire payment?

Reply:
Yeah that sounds good! We might want to have another prefactor, though, because there's still assumptions that `path.hops.len() > 0` scattered around. Will look into that.

Reply:
FYI, two more notes on this approach:
1. We may end up double-booking our own channel liquidity, since we don't know which of our channels is being used inside the blinded path (double-booking liquidity is a general problem for blinded pathfinding anyway, though)
2. ISTM we'll always want to go through the process of selecting additional paths, because the blinded path where we are the intro node may be the most expensive option

For (1), we could decrement each of our available channel balances by the amount used on the path. Not sure it's if worth the additional complexity.

Reply:
I'm gonna address this in a follow-up since this PR is growing. IMO our offers code should still advance blinded paths before pathfinding to them, though adding the behavior you describe is good for keeping `find_route` general-purpose